File: flutter/craftd_widget/lib/presentation/ui/empty/craftd_empty_builder.dart
import 'package:flutter/material.dart';
import 'package:craftd_widget/data/model/empty/empty_properties.dart';
import '../../builder/craftd_builder.dart';
import '../../craftd_view_listener.dart';
import 'craftd_empty.dart';

class CraftDEmptyBuilder extends CraftDBuilder<EmptyProperties> {
  CraftDEmptyBuilder() : super(key: key);

  @override
  Widget craft(EmptyProperties model, CraftDViewListener listener) {
    return const CraftDEmpty();
  }

  @override
  EmptyProperties fromJson(properties) {
    return const EmptyProperties();
  }
  static String key = "CraftDEmpty";
}

File: android_kmp/craftd-compose/src/androidMain/kotlin/com/github/codandotv/craftd/compose/ui/empty/CraftDEmptyBuilder.kt
package com.github.codandotv.craftd.compose.ui.empty

import androidx.compose.runtime.Composable
import com.github.codandotv.craftd.androidcore.data.model.base.SimpleProperties
import com.github.codandotv.craftd.androidcore.presentation.CraftDViewListener
import com.github.codandotv.craftd.compose.builder.CraftDBuilder
class CraftDEmptyBuilder(override val key: String = "") : CraftDBuilder{

    @Composable
    override fun craft(model: SimpleProperties, listener: CraftDViewListener) {
    }
}